client.go

```diff
@@ -361,8 +361,14 @@ func (c *Client) Write(path string, data []byte, _ os.FileMode) error {
 
 // WriteStream writes a stream
 func (c *Client) WriteStream(path string, stream io.Reader, _ os.FileMode) error {
-	// TODO check if parent collection exists
+
+	err := c.createParentCollection(path)
```

Comment:
this adds an extra call instead of an error handling which requires to keep the body ...
that's one reason i left it as TODO ;)

and what is if we add a method which doses createParentCollection + WriteStream?